src/Symfony/Bundle/FrameworkBundle/Tests/Functional/DebugAutowiringCommandTest.php

```diff
@@ -97,4 +97,16 @@ public function testSearchNotAliasedServiceWithAll()
         $tester->run(['command' => 'debug:autowiring', 'search' => 'redirect', '--all' => true]);
         $this->assertStringContainsString('Pro-tip: use interfaces in your type-hints instead of classes to benefit from the dependency inversion principle.', $tester->getDisplay());
     }
+
+    public function testNotConfusedByClassAliases()
+    {
+        static::bootKernel(['test_case' => 'ContainerDebug', 'root_config' => 'config.yml']);
+
+        $application = new Application(static::$kernel);
+        $application->setAutoExit(false);
+
+        $tester = new ApplicationTester($application);
+        $tester->run(['command' => 'debug:autowiring', 'search' => 'ClassAlias', '--all' => true]);
+        $this->assertStringContainsString('Symfony\Bundle\FrameworkBundle\Tests\Fixtures\ClassAliasExampleClass (public)', $tester->getDisplay());
```

Comment:
Before the patch, the output was simple:

```
Symfony\Bundle\FrameworkBundle\Tests\Fixtures\ClassAliasTargetClass
```

It would resolve `ClassAliasExampleClass` to `ClassAliasTargetClass`. To make matters worse, because `ClassAliasTargetClass` is not an alias name in the system (it's not even a service in the system), it doesn't print the service id (or show up unless you pass `--all`) because it looks (incorrectly) like a concrete service.